api/src/main/java/org/pmiops/workbench/api/DataSetController.java

```diff
@@ -291,159 +248,64 @@ private void formatTimestampValues(List<DataSetPreviewValueList> valuePreviewLis
             workspaceNamespace, workspaceId, WorkspaceAccessLevel.WRITER);
     workspaceAuthService.validateActiveBilling(workspaceNamespace, workspaceId);
 
-    validateDataSetRequestResources(
-        dbWorkspace.getWorkspaceId(), dataSetExportRequest.getDataSetRequest());
+    String bucketName =
```

Comment:
I moved it into the generateCodeCells function to move the dataset validation closer to the code where the dataset is actually being used. Functionality and security wise, it should be the same aside from a slightly longer response time in the case where validation fails since the following firecloud get workspace call and notebook object construction will happen first.